db.go

```diff
@@ -380,6 +380,7 @@ func Open(opt Options) (db *DB, err error) {
 		vptr.Decode(vs.Value)
 	}
 
+	db.writeCh = make(chan *request, kvWriteChCapacity)
```

Comment:
There was a race condition between this line and the `flushDiscardStats` function. Moving it before starting the `doWrites` go routine seems to have fixed the race.